buildtest/buildsystem/base.py

```diff
@@ -427,19 +420,14 @@ def run(self):
 
         # Create test directory and run folder they don't exist
         self._create_test_folders()
-        testfile = self._write_test()
-        result = self.run_tests(testfile)
+        self._write_test()
+        result = self.run_tests()
         return result
 
-    def run_tests(self, testfile):
+    def run_tests(self):
```

Comment:
So the issue that I have here is that the client (before this change) is flexible to be updated with a different test file to run. That's actually the entire point - you can instantiate it, and then update the file at runtime without needing to start over. I would prefer to keep this ability because I can see it being useful for some custom implementation / use case in the future.

Reply:
I see your point, i didn't realize you intended to import the BuilderBase class directly. It makes sense, ill fix this.